### PR TITLE
script-directory: init at 1.1.0

### PR DESCRIPTION
**Pull request by Janik-Haag**
###### Description of changes
https://github.com/ianthehenry/sd

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/script-directory/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, substituteAll
+, bashInteractive
+, zsh
```
Comment (figsoda) [inline pkgs/tools/misc/script-directory/default.nix:6]:
```suggestion
, installShellFiles
, patsh
```

File: pkgs/tools/misc/script-directory/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    owner = "ianthehenry";
+    repo = "sd";
+    rev = "v${version}";
+    hash = "sha256-X5RWCJQUqDnG2umcCk5KS6HQinTJVapBHp6szEmbc4U=";
+  };
+
+  installPhase = ''
+    install -D --target-directory="$out/bin" sd
+    install -D --target-directory="$out/share/zsh/site-functions" _sd
+  '';
```
Comment (figsoda) [inline pkgs/tools/misc/script-directory/default.nix:24]:
```suggestion
  nativeBuildInputs = [
    installShellFiles
    patsh
  ];

  installPhase = ''
    runHook preInstall

    patsh -f sd
    install -Dt "$out/bin" sd
    installShellCompletion --zsh _sd

    runHook postInstall
  '';
```

File: pkgs/tools/misc/script-directory/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = {
+    description = "A cozy nest for your scripts";
+    homepage = "https://github.com/ianthehenry/sd";
```
Comment (figsoda) [inline pkgs/tools/misc/script-directory/default.nix:28]:
```suggestion
    changelog = "https://github.com/ianthehenry/sd/tree/${src.rev}#changelog";
```

File: pkgs/tools/misc/script-directory/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ janik ];
+  };
```
Comment (figsoda) [inline pkgs/tools/misc/script-directory/default.nix:31]:
```suggestion
    mainProgram = "sd";
  };
```

File: pkgs/tools/misc/script-directory/default.nix (hunk 2 above)
Comment (Janik-Haag) [reply pkgs/tools/misc/script-directory/default.nix]: patsh seems like quite a nice script :D

**Comment by Janik-Haag** [comment]
@figsoda can you explain what you meant with "a wrapper, something like runCommand with ln -s would work" in the nixpkgs matrix channel?

Thanks for the thorough review and help, I learned quite a bit of new things :heart:

**Comment by figsoda** [comment]
I meant outside this package, so you could add something like this to your `environment.systemPackages`:

```nix
runCommand "sdir" { } ''
  mkdir -p $out/bin
  ln -s ${lib.getExe script-directory} $out/bin/sdir
''
```

so you would be able to install and use both `sd` (as is) and `script-directory` (as `sdir`)

**Review by figsoda — APPROVED** [review]

**Pull request merged**